### PR TITLE
[SPARK-47479][SQL] Optimize cannot write data to relations with multiple paths error log

### PR DESCRIPTION
**Pull request by huangxiaopingRD**
### What changes were proposed in this pull request?
Add paths information to the error log

### Why are the changes needed?
Make it easier for users to troubleshoot problems

### Does this PR introduce _any_ user-facing change?
No

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: common/utils/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -5271,7 +5271,7 @@
   },
   "_LEGACY_ERROR_TEMP_1148" : {
     "message" : [
-      "Can only write data to relations with a single path."
+      "Can only write data to relations with a single path, paths are <paths>"
```
Comment (HyukjinKwon) [inline common/utils/src/main/resources/error/error-classes.json:5274]:
```suggestion
      "Can only write data to relations with a single path but given paths are <paths>"
```

**Review by HyukjinKwon — APPROVED** [review]
cc @MaxGekk

File: common/utils/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -5271,7 +5271,7 @@
   },
   "_LEGACY_ERROR_TEMP_1148" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:5272]: Could you add a sub-class to `UNSUPPORTED_INSERT` with a name like `MULTI_PATH`.

File: common/utils/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (huangxiaopingRD) [reply common/utils/src/main/resources/error/error-classes.json]: Done. Thanks for your review. @HyukjinKwon

File: common/utils/src/main/resources/error/error-classes.json (hunk 2 above)
Comment (huangxiaopingRD) [reply common/utils/src/main/resources/error/error-classes.json:5272]: Okay, I'll update it later.

**Review by MaxGekk — COMMENTED** [review]
The test failure seems unrelated to the changes: [Run / Build modules: yarn, connect](https://github.com/huangxiaopingRD/spark/actions/runs/8385041329/job/22963314087#logs)

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @huangxiaopingRD and @HyukjinKwon for review.

**Review by MaxGekk — APPROVED** [review]

**Pull request closed without merging**